### PR TITLE
feat: plugin options

### PR DESCRIPTION
**Pull request by egorio**
Add plugin options to allow plugins.

This PR is not the final solution, plugins will be wrapped to PluginRunner next PR to make it more nice and clear.

### CODE REVIEW

File: packages/migrate/src/migrate.ts [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
-  DiagnosticCheckPlugin,
-  LintCheckPlugin,
-} from '@rehearsal/plugins';
+import { Plugin, RehearsalService } from '@rehearsal/service';
+import { DiagnosticCheckPlugin, DiagnosticFixPlugin, LintPlugin } from '@rehearsal/plugins';
 import { findConfigFile, parseJsonConfigFileContent, readConfigFile, sys } from 'typescript';
-import type { Reporter } from '@rehearsal/reporter';
-import type { Logger } from 'winston';
+import { PluginOptions } from '@rehearsal/service/dist/src/plugin';
+import { LintPluginOption } from '@rehearsal/plugins/dist/src/plugins/lint.plugin';
```
Comment (brendaHuang) [inline packages/migrate/src/migrate.ts:8]: importing from dist folder?

File: packages/migrate/src/migrate.ts [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
+import { DiagnosticPluginOptions } from '@rehearsal/plugins/dist/src/plugins/diagnostic-fix.plugin';
```
Comment (brendaHuang) [inline packages/migrate/src/migrate.ts:9]: same as above

File: packages/migrate/src/migrate.ts [hunk 3]
```diff
@@ -1,17 +1,16 @@
...
+import { DiagnosticCheckPluginOptions } from '@rehearsal/plugins/dist/src/plugins/diagnostic-check.plugin';
```
Comment (brendaHuang) [inline packages/migrate/src/migrate.ts:10]: dist?

File: packages/service/src/plugin.ts [hunk 4]
```diff
@@ -3,20 +3,16 @@ import type { Logger } from 'winston';
...
+  run(fileName: string, options: PluginOptions): PluginResult;
+}
 
-  constructor(service: RehearsalService, reporter: Reporter, logger?: Logger) {
-    this.service = service;
-    this.logger = logger;
-    this.reporter = reporter;
-  }
+export interface PluginOptions {
+  [key: string]: unknown;
```
Comment (brendaHuang) [inline packages/service/src/plugin.ts:12]: Can we add basePath to PluginOptions?

Comment (egorio) [reply packages/service/src/plugin.ts:12]: You can add this option to specific plugin, like for Lint here:
https://github.com/rehearsal-js/rehearsal-js/pull/675/files#diff-3be91568bd3d2f0e95e214fa7b01784fa345decb2da97e9211aaecc92b30bda7R7-R9

but I will think about sending PluginContext (cli, basePath, service, etc) together with PluginOptions next PR.

**Review by brendaHuang — APPROVED** [review]

**Pull request merged**